dali/operators/image/paste/multipaste.h

```diff
@@ -87,112 +115,302 @@ class MultiPasteOp : public StatelessOperator<Backend> {
     return true;
   }
 
-  template<int ndim = 2>
-  bool Intersects(const Coords& anchors1, const Coords& shapes1,
-                  const Coords& anchors2, const Coords& shapes2) const {
-    for (int i = 0; i < ndim; i++) {
-      if (anchors1.data[i] + shapes1.data[i] <= anchors2.data[i]
-          || anchors2.data[i] + shapes2.data[i] <= anchors1.data[i]) {
+  bool Intersects(ivec2 anchors1, ivec2 shapes1,
+                  ivec2 anchors2, ivec2 shapes2) const {
+    for (int i = 0; i < 2; i++) {
+      if (anchors1[i] + shapes1[i] <= anchors2[i]
+          || anchors2[i] + shapes2[i] <= anchors1[i]) {
         return false;
       }
     }
     return true;
   }
 
-  void AcquireArguments(const OpSpec &spec, const Workspace &ws) {
-    const auto &images = ws.Input<Backend>(0);
+  void AcquireInIdxs(const Workspace &ws) {
+    auto nsamples = ws.GetInputShape(0).num_samples();
+    if (in_idx_.HasExplicitValue()) {
+      DALI_ENFORCE(ws.NumInput() == 1,
+                   make_string("If the `in_ids` is specified, the operator accepts exactly one "
+                               "positional input batch. Got: ",
+                               ws.NumInput(), " inputs."));
+      in_idx_.Acquire(spec_, ws, nsamples);
+      for (int i = 0; i < nsamples; i++) {
+        auto paste_count = GetPasteCount(ws, i);
+        for (int j = 0; j < paste_count; j++) {
+          auto paste_idx = in_idx_[i].data[j];
+          DALI_ENFORCE(
+              0 <= paste_idx && paste_idx < nsamples,
+              make_string("The `in_idx` must be in range [0, .., batch_size - 1]. Got in_idx: ",
+                          paste_idx, ". Input batch size is: ", nsamples, "."));
+        }
+      }
+    }
+  }
+
+  // Get the num channels for the output sample from the pasted regions
+  // or if there are none for given sample
+  inline int InferChannelsNum(const Workspace &ws, int sample_num) {
+    int num_channels = -1;
+    if (!in_idx_.HasExplicitValue()) {
+      num_channels = ws.Input<Backend>(0)[sample_num].shape()[spatial_ndim_];
+      for (int i = 1; i < ws.NumInput(); i++) {
+        int input_channels = ws.Input<Backend>(i)[sample_num].shape()[spatial_ndim_];
```

Comment:
I though about it, but I forgot we have those cool lightweight views with type erasure and assumed it would call for making sure it's all in type-switch.

Reply:
Added GetPasteSample.